base/server/src/main/java/org/dogtagpki/server/cli/SubsystemIdGeneratorUpdateCLI.java

```diff
@@ -0,0 +1,310 @@
+        }
+        throw new EBaseException("Update to " + newGenerator + " not supported");
+    }
+
+    protected void updateRequestNumberRangeGenerator(PKISocketFactory socketFactory, LdapConnInfo connInfo,
+            LdapAuthInfo authInfo, DatabaseConfig dbConfig, String baseDN, IDGenerator newGenerator) throws EBaseException {
+        
+        String value = dbConfig.getString(
+                RequestRepository.PROP_REQUEST_ID_GENERATOR,
+                RequestRepository.DEFAULT_REQUEST_ID_GENERATOR);
+        idGenerator = IDGenerator.fromString(value);
+
+        if (newGenerator == IDGenerator.RANDOM && idGenerator != IDGenerator.RANDOM) {
+            logger.debug("Remove request ranges from configuration");
+            dbConfig.remove(DatabaseConfig.MIN_REQUEST_NUMBER);
+            dbConfig.remove(DatabaseConfig.MAX_REQUEST_NUMBER);
+            dbConfig.remove(DatabaseConfig.REQUEST_INCREMENT);
+            dbConfig.remove(DatabaseConfig.REQUEST_LOW_WATER_MARK);
+            dbConfig.remove(DatabaseConfig.REQUEST_CLONE_TRANSFER_NUMBER);
+            dbConfig.remove(DatabaseConfig.REQUEST_RANGE_DN);
+            dbConfig.put(RequestRepository.PROP_REQUEST_ID_GENERATOR, newGenerator.toString());
+            dbConfig.put(RequestRepository.PROP_REQUEST_ID_LENGTH, "128");
+            return;
+        }
+        if (newGenerator == IDGenerator.LEGACY_2 && idGenerator == IDGenerator.LEGACY) {
+            dbConfig.put(RequestRepository.PROP_REQUEST_ID_GENERATOR, newGenerator.toString());
+            dbConfig.put(RequestRepository.PROP_REQUEST_ID_RADIX, Integer.toString(Repository.DEC));
+            return;
+        }
+        throw new EBaseException("Update to " + newGenerator + " not supported");
+    }
+    
+    private void updateRanges(DatabaseConfig dbConfig, LdapBoundConnection conn, String baseDN, String rangeDN, String configEndSerialNumber,
+        String hostName, String securePort) throws Exception{
+
+        LDAPSearchResults ranges = conn.search(rangeDN, LDAPv3.SCOPE_SUB, "(objectClass=pkiRange)", null, false);
+
+        BigInteger lastUsedSerial = BigInteger.ZERO;
+        boolean nextRangeToUpdate = true;
+        // Search for the last range entry. If it is associated to the CA to update or ranges are not defined
+        // then the nextRange is         
+        while (ranges.hasMoreElements()) {
+            LDAPEntry entry = ranges.next();
+            String endRange = entry.getAttribute("endRange").getStringValues().nextElement();
+            String host = entry.getAttribute("host").getStringValues().nextElement();
+            String port = entry.getAttribute("securePort").getStringValues().nextElement();
+            BigInteger next = new BigInteger(endRange, 16);
+            if (lastUsedSerial.compareTo(next) < 0) {
+                lastUsedSerial = next;
+                nextRangeToUpdate = host.equals(hostName) && port.equals(securePort);
+                
+            }
+        }
```

Comment:
IIUC in this loop we go through all ranges and parse the `endRange` as hexadecimal and use that to find the last range entry. However, in the following loop we're replacing the hexadecimal `endRange` with its decimal value. So if we run this command again for another replica, wouldn't the now decimal `endRange` be incorrectly parsed as hexadecimal?

Reply:
The `nextRange` is updated only if last range is associated with the CA doing the generator update so only one CA will do the update and since the other are not creating new range there should be no conflict. In the following loop only the ranges associated with this CA are updated so if last range is not owned by this CA there is no update therefore there should not be any conflict.

Reply:
Here are the ranges in the DS just before the migration:

```
beginRange: 13
endRange: 30
host: primary.example.com

beginRange: 31
endRange: 48
host: secondary.example.com

nextRange: 49
```

So in the DS at least there seems to be no gap. Is it possible to take these values as decimals, then update `CS.cfg` based on these values, but avoid updating the DS completely? Some serial numbers that used to belong to one server might "move" to another server, but since we're stopping all servers, fixing all ranges at once, and restarting the servers again, that might be OK. What do you think?

Reply:
I have inverted the order of the operation. Update the range to decimal of the updated CA and then look for the next range. In case of not selecting the real last range is not a problem. Considering the case in CI but we suppose the primary range is from 13 to 31  and the secondary range is from 31 to 48. The first updated is the primary so it become 19 49 and it is considered as last. The nextRange will be 50. Then the other CA is update and the range is 50 -72 and the next range is updated again to  73. There are no other operation involved so it is not a problem.

Reply:
The thing is we're still mix up radixes in the second loop:
```
LDAPSearchResults ranges = conn.search(rangeDN, LDAPv3.SCOPE_SUB, "(objectClass=pkiRange)", null, false);
String endRange = entry.getAttribute("endRange").getStringValues().nextElement();
BigInteger next = new BigInteger(endRange);  // could be decimal or hexadecimal that has not been converted to decimal
```
While the code might work with the above example, there isn't really a guarantee that it will work fine in all cases.

As discussed in Slack, we should explore some alternatives:
* avoid modifying DS at all
* store the new SSNv2 ranges in a separate subtree (https://github.com/dogtagpki/pki/pull/4885)